mypyc/genops.py

```diff
@@ -365,15 +365,26 @@ def visit_assignment_stmt(self, stmt: AssignmentStmt) -> Value:
 
     def visit_operator_assignment_stmt(self, stmt: OperatorAssignmentStmt) -> Value:
         target = self.get_assignment_target(stmt.lvalue)
+        rreg = self.accept(stmt.rvalue)
 
         if isinstance(target, AssignmentTargetRegister):
-            rreg = self.accept(stmt.rvalue)
             res = self.binary_op(target.register, rreg, stmt.op, stmt.line)
             res = self.coerce(res, target.type, stmt.line)
             self.add(Assign(target.register, res))
             return INVALID_VALUE
+        if isinstance(target, AssignmentTargetIndex):
+            res = self.add(PrimitiveOp([target.base, target.index], list_get_item_op, stmt.line))
+            res = self.binary_op(res, rreg, stmt.op, stmt.line)
+            res = self.coerce(res, target.type, stmt.line)
+            self.add(PrimitiveOp([target.base, target.index, res], list_set_item_op, stmt.line))
+            return INVALID_VALUE
+        if isinstance(target, AssignmentTargetAttr):
+            res = self.add(GetAttr(target.obj, target.attr, stmt.line))
+            res = self.binary_op(res, rreg, stmt.op, stmt.line)
+            res = self.coerce(res, target.type, stmt.line)
+            self.add(SetAttr(target.obj, target.attr, res, stmt.line))
+            return INVALID_VALUE
```

Comment:
The original code was already starting to do this, but this ends up duplicating a lot of `assign_to_target`. I think the nicest way to structure this would be to make a `read_from_target` method that generates code to read the value out of an `AssignmentTarget`